presto-main/src/main/java/com/facebook/presto/cost/TableScanStatsRule.java

```diff
@@ -0,0 +1,119 @@
+
+        TableStatistics tableStatistics = metadata.getTableStatistics(session, tableScanNode.getTable(), constraint);
+        Map<Symbol, SymbolStatsEstimate> outputSymbolStats = new HashMap<>();
+
+        for (Map.Entry<Symbol, ColumnHandle> entry : tableScanNode.getAssignments().entrySet()) {
+            Symbol symbol = entry.getKey();
+            Type symbolType = types.get(symbol);
+            Optional<ColumnStatistics> columnStatistics = Optional.ofNullable(tableStatistics.getColumnStatistics().get(entry.getValue()));
+            outputSymbolStats.put(symbol, columnStatistics.map(statistics -> toSymbolStatistics(tableStatistics, statistics, session, symbolType)).orElse(UNKNOWN_STATS));
+        }
+
+        return Optional.of(PlanNodeStatsEstimate.builder()
+                .setOutputRowCount(tableStatistics.getRowCount().getValue())
+                .setSymbolStatistics(outputSymbolStats)
+                .build());
+    }
+
+    private SymbolStatsEstimate toSymbolStatistics(TableStatistics tableStatistics, ColumnStatistics columnStatistics, Session session, Type type)
+    {
+        return SymbolStatsEstimate.builder()
+                .setLowValue(asDouble(session, type, columnStatistics.getOnlyRangeColumnStatistics().getLowValue()).orElse(NEGATIVE_INFINITY))
+                .setHighValue(asDouble(session, type, columnStatistics.getOnlyRangeColumnStatistics().getHighValue()).orElse(POSITIVE_INFINITY))
+                .setNullsFraction(
+                        columnStatistics.getNullsFraction().getValue()
+                                / (columnStatistics.getNullsFraction().getValue() + columnStatistics.getOnlyRangeColumnStatistics().getFraction().getValue()))
+                .setDistinctValuesCount(columnStatistics.getOnlyRangeColumnStatistics().getDistinctValuesCount().getValue())
+                .setAverageRowSize(columnStatistics.getOnlyRangeColumnStatistics().getDataSize().getValue() / tableStatistics.getRowCount().getValue())
+                .build();
+    }
+
+    private OptionalDouble asDouble(Session session, Type type, Optional<Object> optionalValue)
+    {
+        return optionalValue
+                .map(value -> toStatsRepresentation(metadata, session, type, value))
+                .orElseGet(OptionalDouble::empty);
+    }
+
+    private Constraint<ColumnHandle> getConstraint(TableScanNode node, Expression predicate, Session session, Map<Symbol, Type> types)
+    {
+        DomainTranslator.ExtractionResult decomposedPredicate = DomainTranslator.fromPredicate(
+                metadata,
+                session,
+                predicate,
+                types);
+
+        TupleDomain<ColumnHandle> simplifiedConstraint = decomposedPredicate.getTupleDomain()
+                .transform(node.getAssignments()::get)
+                .intersect(node.getCurrentConstraint());
+
+        return new Constraint<>(simplifiedConstraint, bindings -> true);
```

Comment:
You know what I would like to say here ;)

Reply:
I added the following comment in the code:
> // TODO Use constraint evaluator from https://github.com/prestodb/presto/pull/9871